server/lib/queries.js

```diff
@@ -20,18 +20,8 @@ const generateFXConversionSQL = aggregate => {
     amountColumn = 'SUM("t."netAmountInCollectiveCurrency"")';
   }
 
-  const fxConversion = [];
-  for (const currency in currencies) {
-    fxConversion.push([currency, currencies[currency].fxrate]);
-  }
-
-  let sql = 'CASE ';
-  sql += fxConversion
-    .map(currency => `WHEN ${currencyColumn} = '${currency[0]}' THEN ${amountColumn} / ${currency[1]}`)
-    .join('\n');
-  sql += 'ELSE 0 END';
-
-  return sql;
+  const amount = await convertToCurrency(1, 'USD', currencyColumn);
```

Comment:
I don't understand how this is going to work, from my understanding it is going to call `getFxRate` like this:

> getFxRate(1, 'USD', 't.currency')

Which I guess will return an error?

Am I missing something?

Reply:
<img width="686" alt="Screenshot 2019-11-02 at 6 16 04 AM" src="https://user-images.githubusercontent.com/29008971/68066437-4c344280-fd38-11e9-8bb2-def8e8555e4b.png">

the amount is passed to the callback

Reply:
That piece of code `generateFXConversionSQL` is absolutely wrong. The concern from @Betree was 100% valid.

Reply:
Reviewing this.

Reply:
@znarf I understand where the bugs are in the code.